packages/web-components/src/utils/dom.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * Swap the DOM elements at each index
+ */
+function swapChildren(rows: NodeList, i: number, j: number): void {
+  const parent = rows[0].parentNode;
+  const one = rows[j];
+  const two = rows[i];
+  const temp = one.nextSibling;
+  parent.insertBefore(one, two);
+  parent.insertBefore(two, temp);
+}
+
+function partition(
```

Comment:
Would you mind adding a comment explaining this function? It's the only one in the file without a description.